scripts/withdraw.ts

```diff
@@ -0,0 +1,89 @@
+import { BigNumber, BigNumberish } from 'ethers';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { parseJsonFile } from './utils';
+import type { PartnerConfig } from './changePartnerShares';
+
+type MinimumErc20TokenContract = {
+  balanceOf: (address: string) => Promise<BigNumber>;
+};
+
+const printStatus = async (
+  collectorAddress: string,
+  partners: string[],
+  erc20TokenInstance: MinimumErc20TokenContract
+) => {
+  const collectorBalance = await erc20TokenInstance.balanceOf(collectorAddress);
+
+  console.log(`Collector balance: ${collectorBalance.toNumber()}`);
+  for (const partner of partners) {
+    const balance = await erc20TokenInstance.balanceOf(partner);
+    console.log(`Address ${partner} balance: ${balance.toNumber()}`);
+  }
+};
+
+const defaultTxGas = 200000;
+
+export const withdraw = async (
+  {
+    collectorAddress,
+    partnerConfig,
+    gasLimit = defaultTxGas,
+  }: {
+    collectorAddress: string;
+    partnerConfig: string;
+    gasLimit?: BigNumberish;
+  },
+  hre: HardhatRuntimeEnvironment
+) => {
+  const parsedPartnerConfig = parseJsonFile<PartnerConfig>(
+    partnerConfig
+  ) as PartnerConfig;
+
+  const parsedPartners = await Promise.all(
+    parsedPartnerConfig.partners.map((partnerConfig) => {
+      return partnerConfig.beneficiary;
+    })
+  );
```

Comment:
I don't think we need to use `Promise.all` here, since there are no promises.
```suggestion
  const parsedPartners = parsedPartnerConfig.partners.map(({beneficiary}) => (beneficiary));
```

Reply:
the Promise.all was there to get rid of the PromiseOrValue type, but i changed it anyways